### ISSUE TITLE
Query with filter on nested object field returns empty hits

### ISSUE THREAD

**Issue by devtud** [meilisearch/MeiliSearch#657]
**Meilisearch version 0.10.1**

Any document in my products index look like:

```json
{
    "id": ...,
    "productName": ...,
    "sellers": [
        {
            "sellerName": "Seller 1",
            ...
        },
        {
            "sellerName": "Seller 2",
            ...
        }
    ]
}
```

I need to filter products by seller, so my request looks like:

```bash
curl --get 'http://localhost:7700/indexes/products/search' \
    --data-urlencode 'q=laptop' \
    --data-urlencode 'filters=sellers.sellerName= "Seller 1"'
```

Even if I have many "laptop" products for "Seller 1", the query response is empty.

```bash
{"hits":[],"offset":0,"limit":20,"nbHits":1348,"exhaustiveNbHits":false,"processingTimeMs":10,"query":"laptop"}
```

In **Meilisearch version 0.10.0** I receive an error for the same query:

```bash
{"message":"filter parsing error: parsing error on line 1 at column 1: expected other"}
```

Is this a bug or the feature for filtering nested fields is not supported? Will it be supported soon?

I would also appreciate some advice on how to solve my problem.

Thanks!

**Comment by MarinPostma** [comment]
Hello @devtud,  nested field filtering is not supported by the filtering engine. Filtering in only supported on strings, numbers, booleans and arrays of the formers. There is no short term plan to support those, but @Kerollmops may decide otherwise :)

**Comment by Kerollmops** [comment]
Hey @devtud,

This current not planned, we will add support for [faceted search](https://github.com/meilisearch/MeiliSearch/issues/625) that will allow many new use-cases and will probably help you filter with more granularity.

You will be able to have a `sellers` field containing an array of sellers `["Seller 1", "Seller 2"]` and be able to facet filter them easily. Using facets is way faster for the engine than having to fetch all the filtered documents each time we need to test if any matches a nested field in an object or an array.

**Comment by devtud** [comment]
The alternative solution is fair enough. My initial expectation that this could work was because I could set the field `sellers.sellerName` as searchable. Even when I try to filter by non-existing fields, i got error like:
```bash
{"message":"filter parsing error: parsing error on line 1 at column 1: attribute `blabla` not found, available attributes are: id, productName, sellers.sellerName"}
```

**Comment by MarinPostma** [comment]
Uhm, the error you got in is the first issue is something that we patched in 0.10.1. However it still would not have worked, but this time silently. Let me explain. As meilisearch does not enforce a schema, what setting `sellers.sellerName` to searchable does is telling the engine that this field may exists on a document, and that if it exists somewhere, can be searched upon. What the engine does not know is that `.` means indirection. The result is that you end up asking him to filter upon an attribute that does not exist in any document, thus returning nothing....

**Comment by MarinPostma** [comment]
FYI, facet filtering (#631) has just been merged into master

**Comment by MarinPostma** [comment]
Hey,
We've decided not to implement this feature, it adds complexity that we don't want for the time being. It seems that most for most usecases, flattening would achieve the required result. We may reconsider if we get enough traction, so feel free to comment. Cheers!

**Comment by rhzs** [comment]
> Hey,
> We've decided not to implement this feature, it adds complexity that we don't want for the time being. It seems that most for most usecases, flattening would achieve the required result. We may reconsider if we get enough traction, so feel free to comment. Cheers!

I don't know what is the metric for "enough traction". I am currently evaluating Meilisearch as an alternative for Elasticsearch (as other would be). And this is clearly one the biggest drawback against Elasticsearch. 

I am not sure how we can proceed without the Facet filtering though flattening is the proposed approach. I don't see it is as enough solution that works for most use cases.

**Comment by Bradleykingz** [comment]
Did you manage to find a solution @rhzs? Is this problem able to be solved using faceted search?

**Comment by icnahom** [comment]
![Screenshot_20220309-220403_Chrome](https://user-images.githubusercontent.com/60504285/157517471-6f7085e6-79d5-42ff-9645-ee6ca37e79c3.jpg)
Facing the same issue with this kind of document.

I like meilisearch a lot, but people like me move to an alternative search engine than change their document structure. So this issue is worth reconsidering.

**Comment by curquiza** [comment]
Hello @icnahom!
Thanks for your report. However I'm not sure to understand your issue and how to reproduce it. This GitHub issue is a reallyyyyy old one. Could you open a new issue, specifying your Meilisearch version, the steps to reproduce, and what you expected :)

thanks a lot!

**Comment by irevoire** [comment]
Oh, actually we already chatted on the meilisearch community slack channel; the problem is that we don't handle the nested fields.

**Comment by Stevemoretz** [comment]
> Ok thank you @irevoire for the precision 👌 For people reading this, nested fields might be available in v0.27.0 (will be released in May) @irevoire is working on it currently :)

So will filters and facets work now with dot notation?

**Comment by irevoire** [comment]
Hey @Stevemoretz, yes, the nested fields have been merged for v0.27.0 as expected and have been available since in all upcoming versions! 😁 

Now you can write `attributes.value_id = 2` in your filter, and it’ll work out of the box if `attributes` or `attributes.value_id` are filterable. (And you don’t need to specify anything for meilisearch to understand what is nested)
If you encounter any problem why trying to use them, please fill in a new issue 👍

**Comment by Stevemoretz** [comment]
> Hey @Stevemoretz, yes, the nested fields have been merged for v0.27.0 as expected and have been available since in all upcoming versions! 😁
> 
> Now you can write `attributes.value_id = 2` in your filter, and it’ll work out of the box if `attributes` or `attributes.value_id` are filterable. (And you don’t need to specify anything for meilisearch to understand what is nested) If you encounter any problem why trying to use them, please fill in a new issue 👍

Thank you for responding sounds amazing!!
